packages/dd-trace/src/appsec/rasp/utils.js

```diff
@@ -51,15 +53,18 @@ function handleResult (actions, req, res, abortController, config) {
   if (blockingAction) {
     // Should block only in express
     if (rootSpan?.context()._name === 'express.request') {
-      const abortError = new DatadogRaspAbortError(req, res, blockingAction)
+      const abortError = new DatadogRaspAbortError(req, res, blockingAction, raspRule)
       abortController.abort(abortError)
 
       // TODO Delete this when support for node 16 is removed
       if (!abortController.signal.reason) {
         abortController.signal.reason = abortError
       }
+      return
     }
   }
+
+  updateRaspRuleMatchMetricTags(req, raspRule, false, false)
```

Comment:
don't you also need to call this in line 50 ? `  if (!abortController || abortOnUncaughtException) return
`